dhcpv4/client.go

```diff
@@ -2,23 +2,55 @@ package dhcpv4
 
 import (
 	"encoding/binary"
+	"errors"
 	"net"
 	"syscall"
 	"time"
 
 	"golang.org/x/net/ipv4"
 )
 
+// MaxUDPReceivedPacketSize is the (arbitrary) maximum UDP packet size supported
+// by this library. Theoretically could be up to 65kb.
 const (
-	MaxUDPReceivedPacketSize = 8192 // arbitrary size. Theoretically could be up to 65kb
+	MaxUDPReceivedPacketSize = 8192
 )
 
+var (
+	// BroadcastAddressBytes are the bytes representing net.IPv4bcast.
+	BroadcastAddressBytes = [4]byte{
+		net.IPv4bcast[0],
+		net.IPv4bcast[1],
+		net.IPv4bcast[2],
+		net.IPv4bcast[3],
+	}
+
+	// DefaultReadTimeout is the time to wait after listening in which the
+	// exchange is considered failed.
+	DefaultReadTimeout = 3 * time.Second
+
+	// DefaultWriteTimeout is the time to wait after sending in which the
+	// exchange is considered failed.
+	DefaultWriteTimeout = 3 * time.Second
+)
+
+// Client is the object that actually performs the DHCP exchange. It currently
+// only has read and write timeout values.
 type Client struct {
-	Network string
-	Dialer  *net.Dialer
-	Timeout time.Duration
+	ReadTimeout, WriteTimeout time.Duration
 }
 
+// NewClient generates a new client to perform a DHCP exchange with, setting the
+// read and write timeout fields to defaults.
+func NewClient() *Client {
```

Comment:
I have to backport this to `DHCPv6` too - at the moment I set the timeout in a cumbersome way

Reply:
Done at https://github.com/insomniacslk/dhcp/pull/14